packages/driver/src/reflection/analyzeQuery.ts

```diff
@@ -144,5 +144,13 @@ function walkCodec(
     ctx.imports.add("Range");
     return `Range<${subCodec.tsType}>`;
   }
+  if (codec instanceof MultiRangeCodec) {
+    const subCodec = codec.getSubcodecs()[0];
+    if (!(subCodec instanceof ScalarCodec)) {
+      throw Error("expected multirange subtype to be scalar type");
+    }
+    ctx.imports.add("MultiRange");
+    return `MultiRange<${subCodec.tsType}>`;
```

Comment:
You can have ranges of `LocalDate/Time`, so we also need to check if the subCodec type needs adding to `ctx.imports`

Reply:
This mirrors what we're doing in `Range`, do you think that's also broken? Looking at how `ArrayCodec` is done here, maybe I need to wrap this and the `Range` subcodec in a recursive call to `walkCodec`?

Reply:
Yeah, I think you're right, `Range` also looks like it has the same problem. Either `walkCodec` or since we know the subCodec is going to be a scalar we could just inline the `codec.importedType` check, both ways seem fine to me.

Reply:
I'll stick with `walkCodec` just in case other behavior falls out and that seems to match how we're doing other types with subcodecs. Thanks for catching this!